### PR TITLE
[FIX] web: effect: display rainbowman by default

### PR DESCRIPTION
**Pull request by aab-odoo**
In [1], we rewrote the effect feature in the wowl infrastructure.
However, we lost something in the process: the "type" of effect
(basically "rainbow_man" as it is the only implemented effect)
became a mandatory param, whereas before it simply defaulted to
"rainbow_man" when not set. As a consequence, previously defined
effects in views could crash.

This commit fixes that issue, and reworks a bit the feature by
introducing a registry of effects, which easily allows to add new
effects from the outside. An effect is a function that may return
a Component (and its props) to display.

Most of the diff in the tests is about moving files where they
should be (the effect feature being in /core, the tests must be
in core/ as well, and not depend on the webclient).

[1] 0573acae2306bf5da2005852da9323ddc59e5431

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by aab-odoo** [comment]
@robodoo override=ci/security

the security risk comes from a test

**Comment by ged-odoo** [comment]
@robodoo override=ci/security
just to make sure I have not the right to do it. also, the security risk comes from a test, so it should be ok

File: addons/web/static/src/core/effects/effect_service.js [hunk 1]
```diff
@@ -32,74 +82,15 @@ export const effectService = {
...
-        function add(type, params = {}) {
-            switch (type.replace("_", "").toLowerCase()) {
-                case "rainbowman":
-                    return rainbowMan(params);
-                default:
-                    throw new Error("NON_IMPLEMENTED_EFFECT");
+        function add(params) {
+            const type = params.type || "rainbow_man";
+            const effect = effectRegistry.get(type);
+            const { Component, props } = effect(env, params) || {};
```
Comment (ged-odoo) [inline addons/web/static/src/core/effects/effect_service.js:91]: i dont think this is quite natural...let's talk about this when you have time

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
Hello @odoo/rd-security . Could one of you override the security check? The risk comes from a js test... Thanks

**Comment by mart-e** [comment]
@robodoo override=ci/security
Reason: replicating in a test the existing behaviour of rainbowman

**Pull request closed without merging**